src/patternfly/components/Label/label.scss

```diff
@@ -1,31 +1,271 @@
+  }
+
+  &.pf-m-cyan {
+    --pf-c-label--BackgroundColor: var(--pf-c-label--m-cyan--BackgroundColor);
+    --pf-c-label__text--Color: var(--pf-c-label--m-cyan__text--Color);
+    --pf-c-label__icon--Color: var(--pf-c-label--m-cyan__icon--Color);
+    --pf-c-label--m-outline__content--before--BorderColor: var(--pf-c-label--m-outline--m-cyan__content--before--BorderColor);
+    --pf-c-label__content--link--hover--before--BorderColor: var(--pf-c-label--m-cyan__content--link--hover--before--BorderColor);
+    --pf-c-label__content--link--focus--before--BorderColor: var(--pf-c-label--m-cyan__content--link--focus--before--BorderColor);
+    --pf-c-label--m-outline__content--link--hover--before--BorderColor: var(--pf-c-label--m-outline--m-cyan__content--link--hover--before--BorderColor);
+    --pf-c-label--m-outline__content--link--focus--before--BorderColor: var(--pf-c-label--m-outline--m-cyan__content--link--focus--before--BorderColor);
   }
+
+  &.pf-m-outline {
+    --pf-c-label__content--before--BorderWidth: var(--pf-c-label--m-outline__content--before--BorderWidth);
+    --pf-c-label__content--before--BorderColor: var(--pf-c-label--m-outline__content--before--BorderColor);
+    --pf-c-label--BackgroundColor: var(--pf-c-label--m-outline--BackgroundColor);
+
+    // stylelint-disable selector-no-qualifying-type
+    a.pf-c-label__content {
+      // stylelint-enable selector-no-qualifying-type
+      &:hover {
+        --pf-c-label__content--before--BorderWidth: var(--pf-c-label--m-outline__content--link--hover--before--BorderWidth);
+        --pf-c-label__content--before--BorderColor: var(--pf-c-label--m-outline__content--link--hover--before--BorderColor);
+      }
+
+      &:focus {
+        --pf-c-label__content--before--BorderWidth: var(--pf-c-label--m-outline__content--link--focus--before--BorderWidth);
+        --pf-c-label__content--before--BorderColor: var(--pf-c-label--m-outline__content--link--focus--before--BorderColor);
+      }
+    }
+  }
+
+  .pf-c-button {
+    --pf-c-button--FontSize: var(--pf-c-label__c-button--FontSize);
+    --pf-c-button--PaddingTop: var(--pf-c-label__c-button--PaddingTop);
+    --pf-c-button--PaddingRight: var(--pf-c-label__c-button--PaddingRight);
+    --pf-c-button--PaddingBottom: var(--pf-c-label__c-button--PaddingBottom);
+    --pf-c-button--PaddingLeft: var(--pf-c-label__c-button--PaddingLeft);
+
+    margin-top: var(--pf-c-label__c-button--MarginTop);
+    margin-right: var(--pf-c-label__c-button--MarginRight);
+    margin-bottom: var(--pf-c-label__c-button--MarginBottom);
+    margin-left: var(--pf-c-label__c-button--MarginLeft);
+  }
+}
+
+.pf-c-label,
+.pf-c-label__content {
+  display: inline-flex;
+  align-items: center;
+}
+
+.pf-c-label__content {
+  &::before {
+    position: absolute;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    left: 0;
+    content: "";
+    border: var(--pf-c-label__content--before--BorderWidth) solid var(--pf-c-label__content--before--BorderColor);
```

Comment:
why does this pseudo element need to be on `pf-c-label__content` ? can it just go on label?

Reply:
The hover border treatment can't apply to the label since we don't want the label border to change when it's a link and the user hovers over the close button.